src/sky.cpp

```diff
@@ -202,58 +203,132 @@ void Sky::render()
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+			}
 
-		// Draw far cloudy fog thing blended with skycolor
-		for (u32 j = 0; j < 4; j++) {
-			video::SColor c = cloudyfogcolor.getInterpolated(m_skycolor, 0.45);
-			vertices[0] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, t);
-			vertices[1] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, t);
-			vertices[2] = video::S3DVertex( 1, 0.12, -1, 0, 0, 1, c, o, o);
-			vertices[3] = video::S3DVertex(-1, 0.12, -1, 0, 0, 1, c, t, o);
-			for (video::S3DVertex &vertex : vertices) {
-				if (j == 0)
-					// Don't switch
-					{}
-				else if (j == 1)
-					// Switch from -Z (south) to +X (east)
-					vertex.Pos.rotateXZBy(90);
-				else if (j == 2)
-					// Switch from -Z (south) to -X (west)
-					vertex.Pos.rotateXZBy(-90);
-				else
-					// Switch from -Z (south) to +Z (north)
-					vertex.Pos.rotateXZBy(-180);
+			// Draw bottom half of the world, without smoothing
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = m_sky_color_bottom;
+				vertices[0] = video::S3DVertex(-1, -1.0, -1, 0, 0, 1, c, t, t);
+				vertices[1] = video::S3DVertex( 1, -1.0, -1, 0, 0, 1, c, o, t);
+				vertices[2] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, o);
+				vertices[3] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, o);
+				for (video::S3DVertex &vertex : vertices) {
+					if (j == 0)
+						// Don't switch
+						{}
+					else if (j == 1)
+						// Switch from -Z (south) to +X (east)
+						vertex.Pos.rotateXZBy(90);
+					else if (j == 2)
+						// Switch from -Z (south) to -X (west)
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+			}
+		} else {
+			// Draw middle band of our smooth skybox
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = m_sky_color_bottom;
+				vertices[0] = video::S3DVertex(-1, -0.03, -1, 0, 0, 1, c, t, t);
+				vertices[1] = video::S3DVertex( 1, -0.03, -1, 0, 0, 1, c, o, t);
+				vertices[2] = video::S3DVertex( 1,  0.02, -1, 0, 0, 1, c, o, o);
+				vertices[3] = video::S3DVertex(-1,  0.02, -1, 0, 0, 1, c, t, o);
+				for (video::S3DVertex &vertex : vertices) {
+					if (j == 0)
+						// Don't switch
+						{}
+					else if (j == 1)
+						// Switch from -Z (south) to +X (east)
+						vertex.Pos.rotateXZBy(90);
+					else if (j == 2)
+						// Switch from -Z (south) to -X (west)
+						vertex.Pos.rotateXZBy(-90);
+					else
+						// Switch from -Z (south) to +Z (north)
+						vertex.Pos.rotateXZBy(-180);
+				}
+				driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+			}
+			
+			// Draw top band of our smooth skybox
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = m_sky_color_bottom;
+				video::SColor c_blend = m_sky_color_top;
```

Comment:
I think it's best to keep the original variable names, as either with or without a smooth gradient, the top is skycolor and the bottom is cloudyfogcolor, there's no need to change the varaiable names that the core devs are used to.

Reply:
Essentially, unless there's good reason to, it's better to change as little as possible instead of as much as possible, to make the diff smaller and minimise disruption and confusion caused by changes.

Reply:
Actually, since in MT master we have
`video::SColor cloudyfogcolor = m_bgcolor;`
It seems clearest to change `bgcolor` to `cloudyfogcolor` everywhere. I agree with you that `bgcolor` is unclear and confusing. But please keep `skycolor` unchanged as it is accurate and familiar. So then the names are improved but changes are minimised.